### PR TITLE
Adding support for two additional HyphenationCompoundWordTokenFilter …

### LINKED ISSUE #8796: Provide access to new settings for HyphenationCompoundWordTokenFilter
**Is your feature request related to a problem? Please describe.**
In Lucene two additional settings for HyphenationCompoundWordTokenFilter are added. They should be available in opensearch

**Describe the solution you'd like**
The new parameters should be supported.

**Describe alternatives you've considered**

**Additional context**
See https://github.com/apache/lucene/issues/9231 which adds the settings.

### PR DESCRIPTION
**Pull request by evankielley**
…settings called noSubMatches and noOverlappingMatches

### Description
This change adds support for / exposes two new settings (noSubMatches and noOverlappingMatches) that were added to Lucene's HyphenationCompoundWordTokenFilter class.

### Related Issues
Resolves #8796 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: modules/analysis-common/src/main/java/org/opensearch/analysis/common/AbstractCompoundWordTokenFilterFactory.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ public abstract class AbstractCompoundWordTokenFilterFactory extends AbstractTok
     protected final int minSubwordSize;
     protected final int maxSubwordSize;
     protected final boolean onlyLongestMatch;
+    protected final boolean noSubMatches;
+    protected final boolean noOverlappingMatches;
```
Comment (msfroh) [inline modules/analysis-common/src/main/java/org/opensearch/analysis/common/AbstractCompoundWordTokenFilterFactory.java:54]: Can you push these fields down into `HyphenationCompoundWordTokenFilterFactory`?

As written, this implicitly adds the settings to `DictionaryCompoundWordTokenFilterFactory` too.

Comment (evankielley) [reply modules/analysis-common/src/main/java/org/opensearch/analysis/common/AbstractCompoundWordTokenFilterFactory.java:54]: Good catch. Will do!

**Comment by evankielley** [comment]
Is there anything else I should do for this PR to resolve issue #8796?

**Comment by msfroh** [comment]
Sorry @evankielley  -- I let this get away from me.

Can you please rebase and resolve the merge conflict in the changelog? Then I think it should be good. The codecov failure is (IMO) non-blocking, because this token filter and its neighbors already lacked coverage. (We should add unit tests for the untested token filters, but that's a broader task.)

**Pull request closed without merging**